x/retry/retry.go

```diff
@@ -23,10 +23,29 @@ package retry
 import (
 	"context"
 
+	"github.com/uber-go/tally"
 	"go.uber.org/yarpc/api/transport"
 	"go.uber.org/yarpc/internal/retry"
 )
 
+// Option is an option for Retry Middleware.
+type Option func(*options)
```

Comment:
Exported types shouldn't reference unexported types; it makes the documentation difficult to understand. Instead, prefer

```
type Option interface {
  apply(*options)
}

type optionFunc func(*options)
func (f optionFunc) apply(opts *options) { f(opts) }
```